src/uct/cuda/cuda_ipc/cuda_ipc_iface.c

```diff
@@ -68,18 +68,34 @@ ucs_status_t uct_cuda_ipc_iface_get_device_address(uct_iface_t *tl_iface,
 static ucs_status_t uct_cuda_ipc_iface_get_address(uct_iface_h tl_iface,
                                                    uct_iface_addr_t *iface_addr)
 {
-    *(pid_t*)iface_addr = getpid();
+    uct_cuda_ipc_iface_addr_t *addr = (uct_cuda_ipc_iface_addr_t*)iface_addr;
+    ucs_status_t status;
+    char proc[32];
+
+    addr->pid = getpid();
+
+    uct_ep_get_process_proc_dir(proc, sizeof(proc), addr->pid);
+    status = ucs_sys_get_file_time(proc, UCS_SYS_FILE_TIME_CTIME,
+                                   &addr->starttime);
```

Comment:
Looks like we're using starttime here to populate keepalive_info_t. Why not populate starttime as part of keepalive_create here https://github.com/openucx/ucx/pull/6352/files#diff-6e9feaecde4caffbd288eaf02c7ee17b6d183cbcfa0fcfaf2dff256332387fdfR614? We seem to using ka.starttime for all purposes and not really using cuda_iface_addr_t->starttime for anything other than populating ka.starttime

Reply:
iface_address is available only during `uct_ep_create()`, then we don't have an access to it
I preferred in the current implementation not to save pid/starttime to an additional structure to initialize keepalive in `uct_ep_check()` as we do in UCT/MM transport, because I don't see memory consumption improvements.